module/VuFind/src/VuFind/Controller/OverdriveController.php

```diff
@@ -210,90 +249,36 @@ public function holdAction()
-        } elseif ($action == 'placeHold') {
-            $actionTitleCode = 'od_hold';
-            $email = $this->params()->fromPost('email');
-            $result = $this->connector->placeOverDriveHold($od_id, $email);
-        } elseif ($action == 'cancelHold') {
-            $actionTitleCode = 'od_cancel_hold';
-            $result = $this->connector->cancelHold($od_id);
-        } elseif ($action == 'returnTitle') {
-            $actionTitleCode = 'od_early_return';
-            $result = $this->connector->returnResource($od_id);
-        } elseif ($action == 'getTitle') {
-            $actionTitleCode = 'od_get_title';
-            $this->debug(
-                'Get Title action. Getting downloadlink using' .
-                $this->getServerUrl('overdrive-hold')
-            );
-            $result = $this->connector->getDownloadLink(
-                $od_id,
-                $format,
-                $this->getServerUrl('overdrive-hold')
-            );
-            if ($result->status) {
-                // Redirect to resource
-                $url = $result->data->downloadLink;
-                $this->debug("redirecting to: $url");
-                return $this->redirect()->toUrl($url);
-            }
+        $actions = [
+            'checkoutConfirm' => ['titleCode' => 'od_checkout', 'resMeth' => 'getConfirmCheckoutRes'],
+            'holdConfirm' => ['titleCode' => 'od_hold', 'resMeth' => 'getHoldConfirmRes'],
+            'cancelHoldConfirm' => ['titleCode' => 'od_cancel_hold', 'resMeth' => false],
+            'suspHoldConfirm' => ['titleCode' => 'od_susp_hold', 'resMeth' => false],
+            'editHoldConfirm' => ['titleCode' => 'od_susp_hold_edit', 'resMeth' => false],
+            'editHoldEmailConfirm' => ['titleCode' => 'od_edit_hold_email', 'resMeth' => 'getEditHoldEmailConfRes'],
+            'returnTitleConfirm' => ['titleCode' => 'od_early_return', 'resMeth' => 'getReturnTitleConfirmResult'],
+            'doCheckout' => ['titleCode' => 'od_checkout', 'resMeth' => 'getCheckoutRes'],
+            'placeHold' => ['titleCode' => 'od_hold', 'resMeth' => 'getPlaceHoldRes'],
+            'editHoldEmail' => ['titleCode' => 'od_hold', 'resMeth' => 'getEditHoldEmailRes'],
+            'suspendHold' => ['titleCode' => 'od_susp_hold', 'resMeth' => 'getSuspendHoldRes'],
+            'editSuspendedHold' => ['titleCode' => 'od_susp_hold_edit', 'resMeth' => 'getEditSuspendedRes'],
+            'cancelHold' => ['titleCode' => 'od_cancel_hold', 'resMeth' => 'getCancelHoldRes'],
+            'returnTitle' => ['titleCode' => 'od_early_return', 'resMeth' => 'getReturnTitleRes'],
+            'getTitle' => ['titleCode' => 'od_get_title', 'resMeth' => 'getDownloadTitleRes'],
+        ];
+
+        if (isset($actions[$action])) {
+            $actionTitleCode = $actions[$action]['titleCode'];
+            $result = $actions[$action]['resMeth'] ? $this->{$actions[$action]['resMeth']}() : false;
```

Comment:
Does it make any difference that in your old code, $result defaulted to `null` but now it gets set to `false` instead for situations where no method is used? I think it's likely the two are functionally equivalent for the purposes of your code, but maybe `null` is a more appropriate value for this situation.